src/basic/chase.c

```diff
@@ -651,7 +651,7 @@ int chase(const char *path, const char *root, ChaseFlags flags, char **ret_path,
 
                                 assert(!path_is_absolute(p));
 
-                                q = path_join(root, p + (*p == '.'));
+                                q = path_join(root, p + STR_IN_SET(p, ".", "./"));
```

Comment:
Hmm, I don't like hardcoding such paths here. `./xxx/yyy`, should be normalized too

Reply:
@YHNdnzj Paths like that will never be returned by `chaseat()`. If they were, we'd use `path_simplify()` here. If I remember  correctly @yuwata specifically didn't want to use `path_simplify()` here because we know `chaseat()` will return normalized paths except for `.` and `./`

Reply:
In that case, could you add a comment?

Reply:
This is already documented a few lines earlier

Reply:
> Note that the result of chaseat() may start with "." (more specifically, it may be "." or "./"), and we need to drop "." in that case.

The way it's phrased is not entirely clear. It could mean that the path might starts with `./`...